Protocol file not parsed correctly on windows

https://github.com/beyondstorage/go-service-tar/pull/1/checks?check_run_id=2782564322#step:5:14

```
=== RUN   TestStorage_Stat
    storage_test.go:127: new storage failed tar new_storager: [endpoint: file:C:\Users\RUNNER~1\AppData\Local\Temp\TestStorage_Stat286526883\001\199446694]: unexpected: open C: The system cannot find the file specified.
--- FAIL: TestStorage_Stat (0.00s)
```

--- a/endpoint.go
+++ b/endpoint.go
@@ -39,7 +39,11 @@ func Parse(cfg string) (p Endpoint, err error) {
 		}
 		return NewHTTPS(host, port), nil
 	case ProtocolFile:
-		return NewFile(s[1]), nil
+		// Handle file paths that contains ":" (often happens on windows platform)
+		//
+		// See issue: https://github.com/beyondstorage/go-endpoint/issues/3
+		path := strings.Join(s[1:], ":")
+		return NewFile(path), nil
 	default:
 		return Endpoint{}, &Error{"parse", ErrUnsupportedProtocol, s[0], nil}
 	}

--- a/endpoint_test.go
+++ b/endpoint_test.go
@@ -69,6 +69,12 @@ func TestParse(t *testing.T) {
 			Endpoint{ProtocolFile, "/root/data"},
 			nil,
 		},
+		{
+			"files contains `:`",
+			"file:C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestStorage_Stat286526883\\001\\199446694",
+			Endpoint{ProtocolFile, "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestStorage_Stat286526883\\001\\199446694"},
+			nil,
+		},
 	}
 
 	for _, tt := range cases {

